File: 217-contains-duplicate/217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        boolean result = false;
        Set<Integer> set = new HashSet<Integer>();
        for(int i = 0; i < nums.length; i++){
            if(set.contains(nums[i])){
                result = true;
            } else {
                set.add(nums[i]);
            }
        }
        return result;
    }
}

//Time Complexity: O(n)
//Space Complexity: O(n)

File: 217-contains-duplicate/217-contains-duplicate.py
# for loop the array
# if we cannot add to hashset, then return false
# else we keep adding to the set

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        res = set()
        for n in nums:
            if n in res:
                return True
            else:
                res.add(n)
        return False